src/workspaces/SettingsModal/SettingsModal.tsx

```diff
@@ -0,0 +1,176 @@
+
+  const signal = useCancellation();
+
+  const getFirstBucketLifecycleSetting = (
+    settings: WorkspaceSetting[],
+    printMultipleWarning = false
+  ): BucketLifecycleSetting | undefined => {
+    const bucketLifecycleSettings: BucketLifecycleSetting[] = settings.filter((setting: WorkspaceSetting) =>
+      isBucketLifecycleSetting(setting)
+    ) as BucketLifecycleSetting[];
+    if (bucketLifecycleSettings.length > 0) {
+      if (printMultipleWarning && bucketLifecycleSettings.length > 1) {
+        // eslint-disable-next-line no-console
+        console.log('Multiple bucket lifecycle settings found, displaying only first');
+      }
+      return bucketLifecycleSettings[0];
+    }
+    return undefined;
+  };
+
+  const getDeleteLifecycleRule = (
+    bucketLifecycleSetting: BucketLifecycleSetting | undefined,
+    printMultipleWarning = false
+  ): DeleteBucketLifecycleRule | undefined => {
+    const configRules = bucketLifecycleSetting?.config.rules;
+    const deleteRules: DeleteBucketLifecycleRule[] = configRules?.filter((rule) =>
+      isDeleteBucketLifecycleRule(rule)
+    ) as DeleteBucketLifecycleRule[];
+    if (printMultipleWarning && !!deleteRules && deleteRules.length > 1) {
+      // eslint-disable-next-line no-console
+      console.log('Multiple delete bucket lifecycle rules found, displaying only first');
+    }
+    return !!deleteRules && deleteRules.length >= 1 ? deleteRules[0] : undefined;
+  };
+
+  useEffect(() => {
+    const loadSettings = withErrorReporting('Error loading workspace settings')(async () => {
+      const settings = (await Ajax(signal)
+        .Workspaces.workspaceV2(namespace, name)
+        .getSettings()) satisfies WorkspaceSetting[];
+      setWorkspaceSettings(settings);
+      const bucketLifecycleSetting = getFirstBucketLifecycleSetting(settings, true);
+      if (bucketLifecycleSetting !== undefined) {
+        const rule = getDeleteLifecycleRule(bucketLifecycleSetting, true);
+        if (rule) {
+          setLifecycleRulesEnabled(true);
+          const prefixes = rule.conditions.matchesPrefix;
+          if (prefixes.length === 0) {
+            setPrefixes([suggestedPrefixes.allObjects]);
+          } else {
+            setPrefixes(prefixes);
+          }
+          setLifecycleAge(rule.conditions.age);
+        }
+      }
+    });
+
+    loadSettings();
+  }, [namespace, name, signal]);
+
+  const persistSettings = withErrorReporting('Error saving workspace settings')(async () => {
+    let newSettings: WorkspaceSetting[];
+    if (lifecycleRulesEnabled) {
+      const prefixesOrNone = _.without([suggestedPrefixes.allObjects], prefixes);
+      newSettings = modifyFirstBucketDeletionRule(workspaceSettings || [], lifecycleAge!, prefixesOrNone);
+    } else {
+      newSettings = removeFirstBucketDeletionRule(workspaceSettings || []);
+    }
+    await Ajax().Workspaces.workspaceV2(namespace, name).updateSettings(newSettings);
```

Comment:
@marctalbott I am calling `updateSettings` when the user presses Save, even if nothing has changed. Am I correct that you added code in Rawls to avoid unnecessary updates?

Reply:
Yes, that's correct. Rawls will only change the settings if there's a difference.